contrib/refresh_checksums.mk

```diff
@@ -70,21 +79,46 @@ $(foreach triplet,$(CLANG_TRIPLETS),$(eval $(call checksum_dep,llvm,$(triplet),a
 
 # External stdlibs
 checksum-stdlibs:
-	@-$(MAKE) -C "$(JULIAHOME)/stdlib" checksumall
+	-$(MAKE) $(QUIET_MAKE) -C "$(JULIAHOME)/stdlib" checksumall
 all: checksum-stdlibs
+.PHONY: checksum-stdlibs
 
 # doc unicode data
 checksum-doc-unicodedata:
-	@-$(MAKE) -C "$(JULIAHOME)/doc" checksum-unicodedata
+	-$(MAKE) $(QUIET_MAKE) -C "$(JULIAHOME)/doc" checksum-unicodedata
 all: checksum-doc-unicodedata
+.PHONY: checksum-doc-unicodedata
 
 # Special LLVM source hashes for optional targets
 checksum-llvm-special-src:
-	@-$(MAKE) -C "$(JULIAHOME)/deps" USE_BINARYBUILDER_LLVM=0 DEPS_GIT=0 BUILD_LLDB=1 BUILD_LLVM_CLANG=1 BUILD_CUSTOM_LIBCXX=1 USECLANG=1 checksum-llvm
+	-$(MAKE) $(QUIET_MAKE) -C "$(JULIAHOME)/deps" USE_BINARYBUILDER_LLVM=0 DEPS_GIT=0 BUILD_LLDB=1 BUILD_LLVM_CLANG=1 BUILD_CUSTOM_LIBCXX=1 USECLANG=1 checksum-llvm
 all: checksum-llvm-special-src
+.PHONY: checksum-llvm-special-src
+
+# merge substring project names to avoid races
+pack-checksum-llvm-tools: | pack-checksum-llvm
+pack-checksum-csl: | pack-checksum-compilersupportlibraries
+
+# define how to pack parallel checksums into a single file format
+pack-checksum-%: FORCE
+	@echo making "$(JULIAHOME)/deps/checksums/$*"
+	@cd "$(JULIAHOME)/deps/checksums" && \
+		for each in $$(ls | grep -i '$*'); do \
+			if [ -d $$each ]; then \
+				for type in $$(ls $$each); do \
+					echo $$each/$$type/$$(cat $$each/$$type); \
+					rm $$each/$$type; \
+				done; \
+				rmdir $$each; \
+			fi; \
+		done >> $*
+	@cd "$(JULIAHOME)/deps/checksums" && \
+		sort $* > $*.tmp && \
+		mv $*.tmp $*
 
 # This file is completely phony
-.PHONY: checksum-*
+FORCE:
+.PHONY: FORCE
```

Comment:
Why can't we mark the individual `pack-checksum-*` targets as phony?  Why do we need this indirect method?

Reply:
It just felt much more tedious to enumerate the list than to do it the old traditional way. Also because `.PHONY: pack-checksum-*` isn't really legal (the wildcard is %, not *, but neither would apply to this)